File: Data Structures & Algorithms/eating-bananas/submission-4.py
class Solution:
    def minEatingSpeed(self, pl: List[int], h: int) -> int:
        l=1
        r=max(pl)
        res=r
        while(l<=r):
            m=(l+r)//2
            t=0
            for i in pl:
                t+=i//m
                if(i%m!=0):
                    t+=1
            if(t<=h):
                res=min(res, m)
                r=m-1
            else:
                l=m+1
                
        # res=min(res, m)
        return res

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-4.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size()==0){return 0;}
        if(nums.size()==1){return 1;}
        unordered_set<int> st;
        int res=1;
        int len=0;
        for(int i : nums){st.insert(i);}
        for(int i : st){
            if(st.find(i-1)!=st.end()){
                int n=i;
                len=1;
                while(st.find(i-1)!=st.end()){len++; i--;}
                while(st.find(n+1)!=st.end()){len++; n++;}
                res=max(res, len);
            }
        }
        // res=max(res, len);
        return res;
        
    }
};
